webrtc/src/benchmarkRtc.ts

```diff
@@ -0,0 +1,168 @@
+import { chromium, Browser } from "playwright";
+import { Client } from "./client";
+import { Args } from "./types";
+import { getEncodingsReport, onEncodingsUpdate } from "./encodingReporter";
+
+const frontendAddress = "http://localhost:5005";
+const fakeVideo = "media/sample_video.mjpeg";
+const fakeAudio = "media/sample_audio.wav";
+const ENCODING_REPORT_PERIOD = 5;
+const INBOUD_TRACK_BANDWIDTH = 0.15 + 0.5 + 1.5;
+const OUTBOUND_TRACK_BANDWIDTH = 1.5;
+
+const delay = (s: number) => {
+  return new Promise((resolve) => setTimeout(resolve, 1000 * s));
+};
+
+export const runBenchmark = async (args: Args) => {
+  const client = new Client(args);
+  await client.purge();
+
+  const browsers = await addPeers(args);
+
+  console.log("Started all browsers, running benchmark");
+
+  const encodingReports = [];
+
+  let time = 0,
+    step = ENCODING_REPORT_PERIOD;
+  while (true) {
+    await delay(step);
+    step = Math.min(step, args.duration - time);
+    time += step;
+
+    const report = getEncodingsReport();
+    encodingReports.push({ ...report.toJson(), timestamp: time });
+
+    writeInPlace(
+      `${time} / ${args.duration}s (${Math.floor(
+        time / args.duration,
+      )}), Encodings: ${report.toString()}}`,
+    );
+
+    if (time == args.duration) break;
+  }
```

Comment:
```suggestion
  while (time <= args.duration) {
    await delay(step);
    step = Math.min(step, args.duration - time);
    time += step;

    const report = getEncodingsReport();
    encodingReports.push({ ...report.toJson(), timestamp: time });

    writeInPlace(
      `${time} / ${args.duration}s (${Math.floor(
        time / args.duration,
      )}), Encodings: ${report.toString()}}`,
    );
  }
```

Reply:
This won't work unfortunately, because at some point the step will be equal to 0. (`step = Math.min(step, args.duration - time)`). So we won't exit the loop. And I don't want to call the last print outside the loop.